gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagProcessingEngine.java

```diff
@@ -140,6 +153,11 @@ public void run() {
           dagTask.conclude();
         } catch (Exception e) {
           log.error("DagProcEngineThread encountered exception while processing dag " + dagProc.getDagId(), e);
+          if (KafkaJobStatusMonitor.isThrowableInstanceOf(e, this.nonRetryableExceptions)) {
+            // conclude the lease so that it is not retried, if the dag proc fails with non-transient exception
+            dagTask.conclude();
+            dagManagementStateStore.getDagManagerMetrics().dagProcessingNonRetryableExceptionMeter.mark();
```

Comment:
couldn't this live entirely encapsulated within `DagProc::process`?  that would give a normal return, which means the DPE would next call `DagTask::conclude`

for that approach, initialize the `DagProcFactory` w/ the exception predicate

Reply:
why to pass another parameter to DagProcFactory or individual DagProcs?
We can just catch it in DagProc::process

Reply:
good point.  really I'm just seeking better encapsulation within `DagProc`, so the DPE remains blissfully unaware of what they're doing and hence what exceptions they might throw.

most ideal might be each `DagProc` class to know which of its exceptions are transient.  e.g. a base class method `boolean isTransientException(Throwable t)`, with a common base class version and any derived class extending beyond that (while delegating to the base class impl as a fallback).

if for expedience, you want a small step in the right direction that would be for the factory to be initialized w/ a predicate as I suggested - given it's the one who decides which specific `DagProc` derived class impls to use.

but after consider further, I really think `DagProc::isTransientException` is preferable.